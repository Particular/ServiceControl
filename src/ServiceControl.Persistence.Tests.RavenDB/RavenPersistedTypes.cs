namespace ServiceControl.UnitTests
{
    using System;
    using System.Linq;
    using NUnit.Framework;
    using Particular.Approvals;
    using Raven.Client.Documents.Indexes;
    using ServiceBus.Management.Infrastructure.Settings;
    using ServiceControl.Persistence;

    class RavenPersistedTypes
    {
        [Test]
        public void Verify()
        {
            var allTypes = typeof(Settings).Assembly.GetTypes().Concat(typeof(RavenQueryExtensions).Assembly.GetTypes()).Concat(typeof(EndpointsView).Assembly.GetTypes());

            var documentTypes = allTypes
                .Where(type => typeof(AbstractIndexCreationTask).IsAssignableFrom(type))
                .SelectMany(indexType => indexType.BaseType?.GenericTypeArguments)
                .Distinct();

            var documentTypeNames = string.Join(Environment.NewLine, documentTypes.Select(t => t.AssemblyQualifiedName).OrderBy(x => x));

            Approver.Verify(documentTypeNames);
        }
    }
}