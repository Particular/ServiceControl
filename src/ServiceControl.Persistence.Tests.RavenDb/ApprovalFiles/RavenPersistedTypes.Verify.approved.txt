ServiceControl.Contracts.CustomChecks.CustomCheck, ServiceControl.Persistence, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.EventLog.EventLogItem, ServiceControl.Persistence, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.MessageAuditing.ProcessedMessage, ServiceControl.Persistence, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.MessageFailures.FailedMessage, ServiceControl.Persistence, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.MessageFailures.GroupComment, ServiceControl.Persistence, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.MessageFailures.QueueAddress, ServiceControl.Persistence, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.Operations.FailedAuditImport, ServiceControl.Persistence, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.Operations.FailedErrorImport, ServiceControl.Persistence, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.Persistence.KnownEndpoint, ServiceControl.Persistence, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.Persistence.MessagesViewIndex+SortAndFilterOptions, ServiceControl.Persistence.RavenDb, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.Persistence.RetryBatch, ServiceControl.Persistence, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.Persistence.RetryBatchGroup, ServiceControl.Persistence, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.Recoverability.FailedMessageRetry, ServiceControl.Persistence, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.Recoverability.FailureGroupMessageView, ServiceControl.Persistence, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.Recoverability.FailureGroupView, ServiceControl.Persistence, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.SagaAudit.SagaHistory, ServiceControl.Audit.Persistence.SagaAudit, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null
ServiceControl.SagaAudit.SagaListIndex+Result, ServiceControl.Persistence.RavenDb, Version=4.0.0.0, Culture=neutral, PublicKeyToken=null